build.ps1

```diff
@@ -401,6 +405,28 @@ function Ensure-WindowsSDK {
   }
 }
 
+function Ensure-SwiftToolchain($Arch) {
+  if (Test-Path "$BinaryCache\toolchains\$PinnedToolchain\PFiles64\Swift\Toolchains\0.0.0+Asserts\usr\bin\swiftc.exe" -IsValid) {
```

Comment:
Is `-IsValid` what we want? It looks like it's only a syntactic check.

<img width="660" alt="image" src="https://github.com/compnerd/swift-build/assets/2314287/5ae2f401-23c7-4c67-888c-4f789c568cda">

Reply:
No, this is wrong.